presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -833,4 +849,35 @@ else if (equality.getRight().equals(column) && columns.contains(equality.getLeft
 
         return Optional.empty();
     }
+
+    public static Optional<Symbol> rewriteExpression(Map<Symbol, Expression> assignments, Expression expression)
+    {
+        checkArgument(expression instanceof CoalesceExpression, "The rewrite can only handle CoalesceExpression");
+        List<Expression> expressionOperands = ((CoalesceExpression) expression).getOperands();
+        checkArgument(expressionOperands.stream().allMatch(SymbolReference.class::isInstance), "Expect operands of CoalesceExpression to be SymbolReference");
+        Set<Symbol> symboslInExpression = expressionOperands.stream()
+                .filter(SymbolReference.class::isInstance)
+                .map(Symbol::from)
+                .collect(toImmutableSet());
+        // We are using the property that the result of coalesce from full outer join keys would not be null despite of the order
+        // of the arguments. Thus we extract and compare the symbols of the CoalesceExpression as a set rather than compare the
+        // CoalesceExpression directly.
+        for (Map.Entry<Symbol, Expression> entry : assignments.entrySet()) {
+            if (entry.getValue() instanceof CoalesceExpression) {
+                List<Expression> assignmentOperands = ((CoalesceExpression) entry.getValue()).getOperands();
+                if (!assignmentOperands.stream().allMatch(SymbolReference.class::isInstance)) {
+                    return Optional.empty();
+                }
+
+                Set<Symbol> symbolsInAssignment = assignmentOperands.stream()
+                        .filter(SymbolReference.class::isInstance)
+                        .map(Symbol::from)
+                        .collect(toImmutableSet());
+                if (symbolsInAssignment.equals(symboslInExpression)) {
```

Comment:
Is this comment addressed? https://github.com/prestosql/presto/pull/262/files#r258687795